EIPS/eip-4799.md

```diff
@@ -0,0 +1,241 @@
+
+The goal of this ERC is to allow users and developers the ability to define arbitrarily complex ownership agreements on top of NFTs that have already been minted. We accomplish this goal through a wrapping standard. This way, new contracts with innovative ownership structures can be deployed, but they can still leverage the authenticity afforded by established NFT contracts - in the past a wrapping contract meant brand new NFTs with no established authenticity, but with this standard that is no longer the case. Developers can now create contracts that enable any NFT to be used in agreements for rentals, collateralized loans, fractionalized ownership, conditional ownership, and much more.
+
+Prior to this standard, wrapping an NFT inside another contract was the only way to add functionality after deployment of the NFT contract, but this meant losing access to the utility of holding the original NFT. Any application querying for the owner of that NFT would determine the wrapping smart contract to be the owner. Using this standard, applications will have a standardized method of interacting with wrapping contracts so that they can continue to direct their utility to users even when the NFT has been wrapped.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity >=0.8.0 <0.9.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "./IERCXNFT.sol";
+
+/// @title ERC-X Non-Fungible Token Wrapping Standard
+/// @dev See https://eips.ethereum.org/EIPS/eip-[TODO]
+/// Note: the ERC-165 identifier for this interface is [TODO].
+
+interface IERCX is IERC165, IERCXNFT {
+    /// @dev This event emits when an NFT is wrapped by this contract.
+    /// It also emits when the wrapping NFT that represents a particular
+    /// wrapped NFT changes.
+    event Wrap(
+        address indexed wrappedContract,
+        uint256 indexed wrappedTokenId,
+        uint256 indexed tokenId
+    );
+
+    /// @notice Find the tokenId of a wrapped NFT's wrapping NFT
+    /// @param wrappedContract The contract address of a wrapped NFT
+    /// @param wrappedTokenId The tokenId of a wrapped NFT
+    /// @return The tokenId of the wrapping NFT
+    function tokenIdOf(IERCXNFT wrappedContract, uint256 wrappedTokenId)
+        external
+        view
+        returns (uint256);
+}
+```
+
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity >=0.8.0 <0.9.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+interface IERCXNFT is IERC165 {
```

Comment:
Might be worth re-ordering these interfaces so that they are read in the order they are required?

Reply:
Had put the other one first since it's the core proposal, but re-ordered to be read in the order they are required